### ISSUE TITLE
the program does not open

### ISSUE THREAD

**Issue by foxprogs** [unfoldingWord/tc-create-app#1582]
RU GL region does not have access to this file https://td.unfoldingword.org/exports/langnames.json
It turns out only if I use vpn. Therefore, the program cannot be started.
![изображение](https://github.com/unfoldingWord/tc-create-app/assets/30548361/b50f4cea-c9d5-46d5-a410-7621334a8472)

**Comment by jacobwegner** [comment]
Hi @foxprogs.  Could you please provide the error in the console related to this CORS failure?

And could you please try running a test using the link below?

https://www.test-cors.org/#?client_method=GET&client_credentials=true&server_url=https%3A%2F%2Ftd.unfoldingword.org%2Fexports%2Flangnames.json&server_enable=true&server_status=200&server_credentials=false&server_tabs=remote

When I use that site (from a US-based IP address), I see the following:

<img width="846" alt="image" src="https://github.com/unfoldingWord/tc-create-app/assets/629062/2e725033-4d3e-4080-80f3-076977d0fd9b">

**Comment by jacobwegner** [comment]
I also just ran a test from https://create.translationcore.com/ in a Chrome console that succeeded:
```javascript
fetch('https://td.unfoldingword.org/exports/langnames.json');
```
<img width="1695" alt="image" src="https://github.com/unfoldingWord/tc-create-app/assets/629062/ca7d54c1-b75c-4c47-8c47-dc9d79c82227">

**Comment by foxprogs** [comment]
@jacobwegner 
![изображение](https://github.com/unfoldingWord/tc-create-app/assets/30548361/571f3985-41ba-4a97-944b-440ec96cd546)
Is it right?

**Comment by foxprogs** [comment]
@jacobwegner I don't have chrome but I have Brave.
![изображение](https://github.com/unfoldingWord/tc-create-app/assets/30548361/57935928-053d-459f-8f6b-55f7468f969a)

**Comment by jacobwegner** [comment]
@foxprogs Thanks for the screenshots.  So the last screenshot it looks to be a time out error (and I don't see the request in the logs).

If you're able to reproduce the original _CORS_ error from your first comment (https://github.com/unfoldingWord/tc-create-app/issues/1582#issue-1732678316) that might also be hepful.

You mentioned that you can access the file from a VPN.  When the VPN is _disabled_, is there any other kind of proxy or DNS configuration you're using?

Put another way, when the VPN is _disabled_, can you reach the main homepage for translationDatabase:

https://td.unfoldingword.org/

**Comment by jacobwegner** [comment]
I just did another test from my US-based IP in Brave and didn't have any issues loading the URL:

<img width="1138" alt="image" src="https://github.com/unfoldingWord/tc-create-app/assets/629062/7907a6e3-39bb-4e0a-88e2-1cc6dc05a3c9">

**Comment by foxprogs** [comment]
@jacobwegner The first screenshot was taken with firefox. There is also a timeout, if you look at the scale, you can see that more than 90 seconds have passed. It is possible that firefox somehow determines the type of error in a different way, and therefore he wrote that cors. From our region, there is no access not only to this file, but in principle to https://td.unfoldingword.org. I can only open this with vpn. For example, everything works with Georgia.

**Comment by foxprogs** [comment]
@jacobwegner The site does not open not only for me. I tried with the provider at my place, with mobile internet. They can't open from other cities either. Most likely due to sanctions, access to the resource is somehow blocked.

**Comment by yakob-aleksandrovich** [comment]
@foxprogs Can you check if the whole unfoldingword.org domain blocked, or just the TD host? Does www.unfoldingword.org open?

**Comment by foxprogs** [comment]
> @foxprogs Can you check if the whole unfoldingword.org domain is blocked, or just the TD host? Does [www.unfoldingword.org](http://www.unfoldingword.org) open?

@yakob-aleksandrovich TD host only

**Comment by yakob-aleksandrovich** [comment]
@jacobwegner Thanks for figuring out it had nothing do with CORS, but with a timeout. We'll have to work around the blockage, somehow, probably we'll need to setup the file at a different location, which requires a change to tcCreate.

**Comment by foxprogs** [comment]
@jacobwegner @yakob-aleksandrovich Thank you. I'm sorry that I didn't accurately describe the error. It was necessary to me to specify at once that does not open because of a timeout.

**Comment by jacobwegner** [comment]
@foxprogs and @yakob-aleksandrovich: Another thing to try is with a different host name:

https://translation-database.herokuapp.com/

https://www.test-cors.org/#?client_method=GET&client_credentials=false&server_url=https%3A%2F%2Ftranslation-database.herokuapp.com%2Fexports%2Flangnames.json&server_enable=true&server_status=200&server_credentials=false&server_tabs=remote

<img width="860" alt="image" src="https://github.com/unfoldingWord/tc-create-app/assets/629062/cad79aaf-d67a-4d51-9d87-72b08b999320">

**Comment by foxprogs** [comment]
@jacobwegner 
![изображение](https://github.com/unfoldingWord/tc-create-app/assets/30548361/d7e62d9f-fe34-4f8e-b7ba-edb2b68e063d)

**Comment by foxprogs** [comment]
And this domain works great https://translation-database.herokuapp.com/

**Comment by abelpz** [comment]
@elsylambert @birchamp I think this is fixed.

**Comment by yakob-aleksandrovich** [comment]
We have td2 and td3 available, of course.
But a better option is Rich's solution: https://qa.door43.org/api/v1/languages/langnames.json
This basically pulls it through the DCS API system. 
 More info: https://unfoldingword.zulipchat.com/#narrow/stream/311721-Infrastructure---ST/topic/td.2Eunfoldingword.2Eorg/near/393731146

**Comment by yakob-aleksandrovich** [comment]
On top of that, it would be good to have this
- configurable through an env/build variable in tC Create
- be able to loop through various options, checking for timeouts (or would that be overkill)?

**Comment by richmahn** [comment]
> We have td2 and td3 available, of course. But a better option is Rich's solution: https://qa.door43.org/api/v1/languages/langnames.json This basically pulls it through the DCS API system. More info: https://unfoldingword.zulipchat.com/#narrow/stream/311721-Infrastructure---ST/topic/td.2Eunfoldingword.2Eorg/near/393731146

Yes, this is now available in production https://git.door43.org/api/v1/languages/langnames.json
